### ISSUE TITLE
[FEATURE] Improve the usability of the actual value selection process

### ISSUE THREAD

**Issue by h1dden-da3m0n** [jenkinsci/rest-list-parameter-plugin#22]
**Is your feature request related to a problem? Please describe.**

This is already helped by #21, but can be further improved by utilizing the `f:combobox` Jenkins UI control or a alternative.

This is related to and a sub-task of #20 

**Describe the solution you'd like**

The user should be able to filter the values offered in the sector of the parameter (e.g. a Combo-box / editable select box).

**Describe alternatives you've considered**

A alternative solution could be a separate filter text-box next to the parameters select-box that can filter the values provided by the select-box.

**Additional context**

As I am no expert in fronted development (ESPECIALLY pure JavaScript) I can most likely only provide this functionality via the `f:combobox`, so any reference to a alternate solution is welcome. (some have already been provided by @ianwilliams1 [here](https://github.com/jenkinsci/rest-list-parameter-plugin/issues/19#issuecomment-711444592))

**Comment by ianwilliams1** [comment]
You may want to keep an eye on [JENKINS-62024](https://issues.jenkins.io/browse/JENKINS-62024) and [jenkins/pull/4688](https://github.com/jenkinsci/jenkins/pull/4688). That implementation will probably satisfy the request.

**Comment by h1dden-da3m0n** [comment]
Thank you for the heads up :+1: Unfortunately, given the activity in that PR, it might take some time for the core team to merge it.
I might have a look and see if I can take anything from it and build my own version :thinking:

**Comment by h1dden-da3m0n** [comment]
@ianwilliams1, sorry for the ping, but would be something like [a `datalist` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist) be enough to fulfill this requirement in your opinion?
I finally have some time on my hands again and am trying to get this in whilst preparing for the new `div` based UI of the core.

**Comment by ianwilliams1** [comment]
Based on the behaviour of the example, it appears to satisfy the request "simply start typing; matching values should filter the list.".
Only constraint seems to be does not work properly in IE / Edge<https://stackoverflow.com/q/60038213>.

Maybe email the "Mailing list for developers of Jenkins" jenkinsci-dev@googlegroups.com<mailto:jenkinsci-dev@googlegroups.com> and see what guidance they are providing?

**Comment by h1dden-da3m0n** [comment]
I knew about the IE and legacy Edge limitations, but good call with the mailing list I will give that one a shot.

**Comment by h1dden-da3m0n** [comment]
Oh boy. Warning, this might sound like a rant. (there is a TLDR at the end)

I have now "played" around with this for about 2 days, to get absolutely nowhere :frowning_face:.
That what I have got working is unusable from a UX perspective and I am not going to start commenting on my mess of a code (JS code mind you).
I am pulling the emergency shoot right now before I go into a meltdown over how much I dislike doing web front ends without good documentation or a supportive framework (e.g. Angular, Vue, ...).
DISCLAIMER: I not solely ended using `datalist` due to UX problems that even I could not bear with and tried several approaches to "improve it", a lot of building, deploying and a lot of me cussing over my own JS misery later and that what I got working is in no way better than what I started with.

**TLDR**:
I give up on using `datalist` or trying to implement a 'combobox' component on my own (no idea why I tried the latter) to save my sanity/motivation to keep going.

**Next steps**:
Well, as far as I can tell I have 2 options (and I don't particularly like either, despite preferring one over the other due to UX and design):

1. continue using a simple select component and add a filter input field that gets used with some JS to filter the select options.  This should be a very straight forward solution, however I am no fan for the UI nor the UX of this solution.
2. use a 3rd party solution (obviously Open Source) that offers a proper filterable select component (e.g. [Select2](https://github.com/select2/select2)). This approach should be as easy as it can be and I like its UI/UX more. However, I am unsure if this is an acceptable approach or would be playing along well with the Jenkins UI as a whole.

Regardless, I will take a break of this task till the weekend and select one of these approaches (most likely going to try both anyway). If you have any comment or input on this feel free to share it, I would definitively appreciate a second opinion.

**Comment by h1dden-da3m0n** [comment]
![image](https://user-images.githubusercontent.com/33120068/108920538-14d85c00-7635-11eb-9eac-5224ad352990.png)

After long delay, and a couple of times I got distracted by other things, here a sneak peak on what i have (yes I am using `select2`)
